test/test_dense/test_construction.cc

```diff
@@ -336,3 +336,100 @@ TEST_F(TestFillAxis, long_axis_2x4) {
 TEST_F(TestFillAxis, short_axis_2x4) {
     auto_transpose<float>(2, 4, RandBLAS::MajorAxis::Short);
 }
+
+class TestStateUpdate : public ::testing::Test
+{
+    protected:
+
+    virtual void SetUp(){};
+
+    virtual void TearDown(){};
+
+    template <typename T>
+    static void test_var_mat_gen(
+        uint32_t key,
+        int64_t n_rows,
+        int64_t n_cols,
+        RandBLAS::DenseDistName dn
+    ) {
+        // Allocate workspace
+        int64_t size = n_rows * n_cols;
+        std::vector<T> A(size, 0.0);
+        std::vector<T> B(size, 0.0);
+
+        // Construct the sketching operator
+        RandBLAS::DenseDist D = {
+            .n_rows = n_rows,
+            .n_cols = n_cols,
+            .family = dn
+        };
+
+        auto state = RandBLAS::RNGState(key);
+        auto next_state = RandBLAS::fill_dense(D, A.data(), state);
+        RandBLAS::fill_dense(D, B.data(), next_state);
+
+        ASSERT_TRUE(!(A == B));
+    }
+
+    template <typename T>
+    static void test_identity(
+        uint32_t key,
+        int64_t n_rows,
+        int64_t n_cols,
+        RandBLAS::DenseDistName dn
+    ) {
+        // Allocate workspace
+        int64_t size = n_rows * n_cols;
+        std::vector<T> A(size, 0.0);
+        std::vector<T> B(size, 0.0);
+
+        // Construct the sketching operator
+        RandBLAS::DenseDist D1 = {
+            .n_rows = n_rows,
+            .n_cols = n_cols / 2,
+            .family = dn
+        };
+
+        RandBLAS::DenseDist D2 = {
+            .n_rows = n_rows,
+            .n_cols = n_cols,
+            .family = dn
+        };
```

Comment:
With the way that this test is currently being called, we have ``n_rows == n_cols``. That's a problematic edge-case when it comes to determining short-axis major versus long-axis major layout. We should first get this working with ``n_cols < n_rows``.